vnext/Desktop/Modules/WebSocketModule.cpp

```diff
@@ -148,26 +148,42 @@ std::shared_ptr<IWebSocketResource> WebSocketModule::GetOrCreateWebSocket(int64_
   {
     auto ws = IWebSocketResource::Make(std::move(url));
     auto weakInstance = this->getInstance();
-    ws->SetOnError([this, id, weakWs = weak_ptr<IWebSocketResource>(ws), instance = weakInstance.lock()](const IWebSocketResource::Error& err)
+    ws->SetOnError([this, id, weakWs = weak_ptr<IWebSocketResource>(ws), weakInstance](const IWebSocketResource::Error& err)
     {
+      auto strongInstance = weakInstance.lock();
+      if (!strongInstance)
+        return;
+
       auto strongWs = weakWs.lock();
```

Comment:
Not sure if this is new code, but we need to be null-checking this as well before using the `this` ptr.

Reply:
Not following. in this code block, the pointer is being null-checked before using `this`.
What usage of `this` do you refer to?

```
      auto strongInstance = weakInstance.lock();
      if (!strongInstance)
        return;

      auto strongWs = weakWs.lock();
      auto args = dynamic::object("id", id);
      this->SendEvent("websocketOpen", std::move(args));
```

Reply:
`strongWs` will be null if `weakWs` has died. Locking here won't really do anything.

I mistead at first and thought `strongWs` referred to `this` but was incorrect. I can see that's for `instance`. We should be safe then I thing, but I'm still confused by locking strongWs.

Reply:
I kept that reference to ensure the lifetime of the lambdas (they are ultimately owned by the WebSocket resource (`strongWs`)).

Is this unnecessary?

Reply:
Granted, the variable should definitely be renamed to `resource`|`weakResource`|`strongResource`.

Reply:
Given we're passing as a weak_ptr right now, this won't keep anything alive. We wouldn't want to keep it alive though since that would make a cyclical reference. I think the only way the callback would be called would be if the resource is alive to begin with though, so you should be fine without.